dds/DCPS/PeriodicTask.cpp

```diff
@@ -0,0 +1,67 @@
+/*
+ * Distributed under the OpenDDS License.
+ * See: http://www.opendds.org/license.html
+ */
+
+#include "DCPS/DdsDcps_pch.h" //Only the _pch include should start with DCPS/
+
+#include "PeriodicTask.h"
+
+#include "Timers.h"
+
+OPENDDS_BEGIN_VERSIONED_NAMESPACE_DECL
+
+namespace OpenDDS {
+namespace DCPS {
+
+void PeriodicTask::enable(bool reenable, const TimeDuration& period)
+{
+  {
+    ACE_GUARD(ACE_Thread_Mutex, g, mutex_);
+    user_enabled_ = true;
+  }
+  RcHandle<ReactorInterceptor> interceptor = interceptor_.lock();
+  if (interceptor) {
+    interceptor->execute_or_enqueue(make_rch<ScheduleEnableCommand>(rchandle_from(this), reenable, period));
+  }
+}
+
+void PeriodicTask::disable()
+{
+  {
+    ACE_GUARD(ACE_Thread_Mutex, g, mutex_);
+    user_enabled_ = false;
+  }
+  RcHandle<ReactorInterceptor> interceptor = interceptor_.lock();
+  if (interceptor) {
+    interceptor->execute_or_enqueue(make_rch<ScheduleDisableCommand>(rchandle_from(this)));
+  }
+}
+
+void PeriodicTask::enable_i(bool reenable, const TimeDuration& per)
+{
+  if (!enabled_) {
+    timer_ = Timers::schedule(reactor(), *this, 0, TimeDuration(), per);
+    if (timer_ == Timers::InvalidTimerId) {
+      ACE_ERROR((LM_ERROR, "(%P|%t) PeriodicTask::enable_i"
+                  " failed to schedule timer %p\n", ACE_TEXT("")));
```

Comment:
Since we're chancing these, we should convert these to use `LogLevel::Error`, there are a few more in `Multitask.cpp`.